hedvig-market/src/main/kotlin/com/hedvig/android/market/MarketManager.kt

```diff
@@ -1,51 +1,56 @@
 package com.hedvig.android.market
 
-import android.annotation.SuppressLint
-import android.content.Context
-import androidx.preference.PreferenceManager
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.runBlocking
 
 interface MarketManager {
   val enabledMarkets: List<Market>
-  var market: Market?
+  val market: Market?
+  suspend fun setMarket(market: Market)
+  suspend fun removeMarket()
+  fun observeMarket(): Flow<Market?>
 }
 
 internal class MarketManagerImpl(
-  context: Context,
   isDebug: Boolean,
+  private val dataStore: DataStore<Preferences>,
 ) : MarketManager {
 
-  private val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
-
   override val enabledMarkets = listOfNotNull(
     Market.SE,
     Market.NO,
     Market.DK,
     if (isDebug) Market.FR else null,
   )
+  override val market: Market?
+    get() = runBlocking { observeMarket().first() }
 
-  override var market: Market?
-    get() = getMarketLocally()
-    set(value) {
-      value?.let(::setMarketLocally) ?: removeMarket()
+  override fun observeMarket(): Flow<Market?> {
+    return dataStore.data.map { preferences ->
+      val market = preferences[marketPreferenceKey] ?: return@map null
+      Market.valueOf(market)
     }
+  }
 
-  private fun getMarketLocally(): Market? {
-    return sharedPreferences
-      .getString(Market.MARKET_SHARED_PREF, null)
-      ?.let { Market.valueOf(it) }
+  override suspend fun setMarket(market: Market) {
+    dataStore.edit { preferences ->
+      preferences[marketPreferenceKey] = market.name
+    }
   }
 
-  @SuppressLint("ApplySharedPref") // We need to do this right away
-  private fun setMarketLocally(market: Market) {
-    sharedPreferences.edit()
-      .putString(Market.MARKET_SHARED_PREF, market.name)
-      .commit()
+  override suspend fun removeMarket() {
+    dataStore.edit { preferences ->
+      preferences.remove(marketPreferenceKey)
+    }
```

Comment:
I think we may have had some cases where removing the market completely makes the marketing screen act a bit weird. Could be wrong though. Just gonna have to make sure that when this happens, we are also logged out for sure (there's no guard-rails for this yet, but I see we're only calling this on logout() so let's make sure we keep it that way) otherwise a lot of weirdness will happen.
And that the marketing activity handles this case well, which I do not really know if it does ATM, would need to check.

Reply:
Good point. I think we could remove this functionality altogether tbh.

Reply:
Yeah, defaulting to SE is the safe bet usually. But this does potentially come with the side-effect that in the cases where things are actually going wrong, for non-SE members it may break completely. But handling the no-market case in the marketing screen should be a separate PR regardless, if we ever want/need to do it.